test/browser/test_glfw3_css_scaling.c

```diff
@@ -0,0 +1,119 @@
+ * - Being undocumented/untested, this feature was accidentally removed when HiDPI support was added (in 3.1.51 / December 2023).
+ * - This test was added to document this feature and ensure proper behavior.
+ *
+ * What does this feature do?
+ * - if there is a CSS rule that specifies the size of the canvas (ex: `#canvas { width: 100%;}`), then the canvas
+ *   will be scaled to match this value. Note that from a GLFW point of view, the size of the window remains what
+ *   gets specified in `glfwCreateWindow` and/or `glfwSetWindowSize`
+ * - only global CSS rules apply, as setting a CSS rule on the canvas itself is removed (ex: `<canvas style="width:100%;">` is ignored)
+ *
+ * In HiDPI mode, (`glfwWindowHint(GLFW_SCALE_TO_MONITOR, GLFW_TRUE)`), this feature cannot work and as a result is disabled (because the canvas
+ * is sized using `devicePixelRatio` as the factor and CSS is used to scale it at the desired size).
+ */
+
+/**
+ * As explained above, this function adds a global CSS rule instead of setting the CSS style directly on the canvas
+ * because it gets ignored otherwise. */
+EM_JS(void, addCSSScalingRule, (), {
+  const style = document.createElement('style');
+  style.appendChild(document.createTextNode("#canvas { width: 700px; height: 500px; }"));
+  document.head.appendChild(style);
+})
+
+/**
+ * Since it is unclear in which browser/resolution this test will run we compute the actual ratio used by the test.
+ * This has the neat effect that the test can be run manually on a HiDPI screen.
+ */
+EM_JS(double, getDevicePixelRatio, (), {return (typeof devicePixelRatio == 'number' && devicePixelRatio) || 1.0;})
+
+/**
+ * Checks window size and framebuffer size according to ratio */
+static void checkWindowSize(GLFWwindow *window, int expectedWidth, int expectedHeight, float ratio) {
+  // first check the window size
+  int w, h;
+  glfwGetWindowSize(window, &w, &h);
+  printf("windowSize => %d == %d && %d == %d\n", w, expectedWidth, h, expectedHeight);
+  assert(w == expectedWidth && h == expectedHeight);
+
+  // second check the frame buffer size
+  int fbw, fbh;
+  glfwGetFramebufferSize(window, &fbw, &fbh);
+  printf("framebufferSize => %d == %d && %d == %d\n", fbw, (int) (expectedWidth * ratio), fbh, (int) (expectedHeight * ratio));
+  assert(fbw == (int) (expectedWidth * ratio) && fbh == (int) (expectedHeight * ratio));
+}
+
+int main() {
+
+  assert(glfwInit() == GLFW_TRUE);
+
+  // Use Case #1
+  // Create a window without HiDPI support and without CSS rule => expected sizes to match
+  {
+    printf("Use case #1\n");
+    GLFWwindow* window = glfwCreateWindow(640, 480, "test_glfw3_css_scaling.c", NULL, NULL);
+    assert(window);
+    checkWindowSize(window, 640, 480, 1.0);
+    double w, h;
+    emscripten_get_element_css_size("#canvas", &w, &h);
+    printf("CSS Size=%.0fx%.0f\n", w, h);
+    assert(w == 640 && h == 480);
+    glfwDestroyWindow(window);
+  }
```

Comment:
How about putting each of these test cases in its own C function so the main function looks like:

```
int main() {
  assert(glfwInit() == GLFW_TRUE);
  test_foo();
  test_bar();
  test_baz();
  printf("All tests complete\n");
  glfwTerminate();
  return 0;
}
```

This form has a few advantaged I think.